pyqtgraph/exporters/CSVExporter.py

```diff
@@ -36,23 +34,48 @@ def export(self, fileName=None):
         data = []
         header = []
 
-        appendAllX = self.params['columnMode'] == '(x,y) per plot'
+        appendAllX = self.params['columnMode'] == '(x,y,err_y,err_x) per plot'
 
         for i, c in enumerate(self.item.curves):
             cd = c.getData()
             if cd[0] is None:
                 continue
             data.append(cd)
+
             if hasattr(c, 'implements') and c.implements('plotData') and c.name() is not None:
                 name = c.name().replace('"', '""') + '_'
                 xName, yName = '"'+name+'x"', '"'+name+'y"'
             else:
                 xName = 'x%04d' % i
                 yName = 'y%04d' % i
+                e_yName = 'e_y%04d' % i
+                e_xName = 'e_x%04d' % i
+
             if appendAllX or i == 0:
-                header.extend([xName, yName])
+                header.extend([xName, yName,e_yName,e_xName ])
             else:
-                header.extend([yName])
+                header.extend([yName,e_yName,e_xName ])
+
+        for indx, item in enumerate(self.item.items):
+
+            if self.item.items[indx].__class__.__name__ == "InfiniteLine":                   
```

Comment:
instead of checking for classname and doing matchup, you can do ....

```python
from ..GraphicsItems import InfiniteLine, FillBewteenItem, TextItem

...


if isinstance(self.item.items[indx], (InfiniteLine, FillBetweenItem, TextItem)):
    continue
```